packages/ui/src/proposals/hooks/useProposals.ts

```diff
@@ -0,0 +1,22 @@
+import { mockProposals } from '@/mocks/data/mockProposals'
+import { useMockDelay } from '@/mocks/hooks/useMockDelay'
+
+import { Proposal } from '../types'
+
+const mock: { [k: string]: { proposals: Proposal[] } } = {
+  current: { proposals: mockProposals },
+  past: { proposals: [] },
+}
+
+export const useProposals = ({ type }: { type: 'current' | 'past' }) => {
```

Comment:
```suggestion
const mock: { [k: string]: { proposals: Proposal[] } } = {
  current: { proposals: mockProposals },
  past: { proposals: [] },
}

interface Props {
  type: 'current' | 'past'
}

export const useProposals = ({ type }: Props) => {
```

Like this: :point\_up: Then you can extend the interface if needed and the function is easier to read.